src/main/kotlin/domain/MoveStrategy.kt

```diff
@@ -0,0 +1,16 @@
+package domain
+
+import java.util.*
+
+private val random = Random()
+private const val CRITERIA = 4
+
+enum class MoveStrategy(val description: String, private val function: () -> Int) {
+    RANDOM("랜덤하게 가거나, 혹은 가지 않는 전략", { random.nextInt() }),
```

Comment:
```suggestion
interface Strategy {
    val number: Int
}

enum class MoveStrategy(val description: String) : Strategy {
    RANDOM("랜덤하게 가거나, 혹은 가지 않는 전략") {
        override val number: Int
            get() = random.nextInt()
    },
```
이렇게도 쓸 수 있어요(취향)
뭘 많이 녹여내볼라고 했는데
1. 인터페이스에 `val`로 선언된 프로퍼티가 들어감
1. 이넘의 인터페이스 구현
1. 생성자는 마지막 람다를 바깥으로 못빼지만 이넘은 이런 식으로는 가능함